### PR TITLE
Refactor: COMMENT section

### PR DESCRIPTION
**Pull request by tokitouq**

### CODE REVIEW

File: backend/django_core/static_src/constants/device.ts
Comment (tokitouq) [inline backend/django_core/static_src/constants/device.ts:1]: @baseplate-admin check if this looks good to you.
well this wont update on `resize`

Comment (baseplate-admin) [reply backend/django_core/static_src/constants/device.ts:1]: Uh no

Did you forget about sm md?

File: backend/django_core/static_src/components/minor/Comment/CommentBlock.svelte [hunk 1]
```diff
@@ -231,7 +243,7 @@
         {/if}
 
         <!-- Render replies here -->
-        {#if item.childrens !== 0 && !item.collapse}
+        {#if !_.isEmpty(item.child) && item.childrens !== 0 && !item.collapse}
             <div class="mt-5 flex flex-col gap-5 md:mt-[1.5vw] md:gap-[1.5vw]">
                 {#each item.child as comment, index}
                     <svelte:self
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/minor/Comment/CommentBlock.svelte:249]: You see there's a `<svelte:self></svelte:self>` tag here?

You should have attached the dispatcher here too

File: backend/django_core/static_src/components/minor/Comment/CommentBlock.svelte [hunk 2]
```diff
@@ -100,9 +100,9 @@
         }
     };
 
-    const handle_more_click = async () => {
+    const handle_more_click = async (path: string) => {
         dispatch("moreComments", {
-            path: item.path,
+            path: path,
         });
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/minor/Comment/CommentBlock.svelte]: Reasoning behind this?

File: backend/django_core/static_src/components/minor/Comment/CommentBlock.svelte [hunk 3]
```diff
@@ -100,9 +100,9 @@
         }
     };
 
-    const handle_more_click = async () => {
+    const handle_more_click = async (path: string) => {
         dispatch("moreComments", {
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/minor/Comment/CommentBlock.svelte:104]: `moreComments` -> `more_comments`

File: backend/django_core/static_src/components/minor/Comment/json_to_tree.ts [hunk 4]
```diff
@@ -7,14 +7,40 @@ export class JSONToTree {
     constructor({ json, old_json, specific_path }:
         { json: Comment[]; old_json?: Comment[], specific_path?: string }
     ) {
-        if (old_json) {
+        if (old_json && specific_path) {
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/minor/Comment/json_to_tree.ts:10]: This is a tad bit cumbersome no?

**Review by baseplate-admin — CHANGES_REQUESTED** [review]
Overall i dont get what you are trying to do?

Do you need to mutate our tree? I feel like mutating tree for a large section is not only resource intensive but also time consuming

File: backend/django_core/static_src/functions/tailwind.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+// Code from : https://stackoverflow.com/a/59982143
+import resolveConfig from "tailwindcss/resolveConfig";
+import tailwindConfig from "$tailwind";
+
+export default resolveConfig(tailwindConfig);
```
Comment (baseplate-admin) [inline backend/django_core/static_src/functions/tailwind.ts:5]: This should be modified, we dont want the entire tailwind payload in JS object

File: backend/django_core/static_src/components/minor/Comment/json_to_tree.ts [hunk 6]
```diff
@@ -4,31 +4,64 @@ import * as _ from "lodash-es";
...
+                };
+            };
+        };
+
+        new_branch.forEach((branch) => {
+            merge(branch, old_branch);
+        });
+
+        return old_branch;
+    };
```
Comment (baseplate-admin) [inline backend/django_core/static_src/components/minor/Comment/json_to_tree.ts:43]: See tatttoo?

File: backend/django_core/static_src/functions/tailwind.ts (hunk 5 above)
Comment (tokitouq) [reply backend/django_core/static_src/functions/tailwind.ts:5]: This will be fixed on next PR.
I hope so :")

**Pull request merged**